d2planner/src/calculators.js

```diff
@@ -13,15 +32,30 @@ const calcLookup = {
   par6: createParamCalculator(6),
   par7: createParamCalculator(7),
   par8: createParamCalculator(8),
+  clc1: createCalculatorOnCalcField('calc1'),
+  clc2: createCalculatorOnCalcField('calc2'),
+  clc3: createCalculatorOnCalcField('calc3'),
+  clc4: createCalculatorOnCalcField('calc4'),
   lvl: (skill, lvl, skillLevels) => (lvl),
 };
 
-function calculateToHit(skill, lvl, skillLevels) {
+function calculateToHit (skill, lvl, skillLevels) {
   const toHitExpression = skill.toHitCalc;
   if (toHitExpression) {
     return calculateSkillValue(toHitExpression, skill, lvl, skillLevels);
   }
-  return (skill.toHit || 0) + (skill.levToHit || 0) * (lvl - 1)
+  return (skill.toHit || 0) + (skill.levToHit || 0) * (lvl - 1);
+}
+
+function calculateManaCost (skill, lvl, skillLevels, multiplier=null) {
+  const mana = skill.mana;
+  const lvlMana = skill.lvlMana || 0;
+  const manaShift = skill.manaShift;
+  let cost = (mana + lvlMana * (lvl - 1)) * 2 ** (manaShift - 8);
+  if (multiplier !== null) {
+    cost *= multiplier;
+  }
```

Comment:
Am I right to use `let` here and modify the varaible - or would you just create a second `const` for the multiplied value on the second line?